src/daft-core/src/utils/stats.rs

```diff
@@ -0,0 +1,82 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::{
+        ops::{DaftCountAggable, DaftSumAggable, GroupIndices, VecIndices},
+        prelude::{Float64Array, UInt64Array},
+    },
+    count_mode::CountMode,
+};
+
+#[derive(Clone, Copy, Default, Debug)]
+pub struct Stats {
+    pub sum: f64,
+    pub count: u64,
+    pub mean: Option<f64>,
+}
+
+pub fn calculate_stats(array: &Float64Array) -> DaftResult<Stats> {
+    let sum = array.sum()?.get(0);
+    let count = array.count(CountMode::Valid)?.get(0);
+    let stats = sum
+        .zip(count)
+        .map_or_else(Default::default, |(sum, count)| Stats {
+            sum,
+            count,
+            mean: calculate_mean(sum, count),
+        });
+    Ok(stats)
+}
+
+pub fn grouped_stats<'a>(
+    array: &Float64Array,
+    groups: &'a GroupIndices,
+) -> DaftResult<impl Iterator<Item = (Stats, &'a VecIndices)>> {
+    let grouped_sum = array.grouped_sum(groups)?;
+    let grouped_count = array.grouped_count(groups, CountMode::Valid)?;
+    assert_eq!(grouped_sum.len(), grouped_count.len());
+    assert_eq!(grouped_sum.len(), groups.len());
+    Ok(GroupedStats {
+        grouped_sum,
+        grouped_count,
+        groups: groups.iter().enumerate(),
+    })
+}
+
+struct GroupedStats<'a, I: Iterator<Item = (usize, &'a VecIndices)>> {
+    grouped_sum: Float64Array,
+    grouped_count: UInt64Array,
+    groups: I,
```

Comment:
Small nit but why not zip all these together or have the grouped_sum / grouped_count be iterators as well?

Reply:
Hmm I tried this one out. I was running into lifetime issues.

<img width="898" alt="image" src="https://github.com/user-attachments/assets/8c7b0700-70a7-474a-b511-09eb3ebb725a">

I think the reason why is because `grouped_stats` owns `grouped_sum` and `grouped_count`. But when I call `grouped_sum.into_iter()`, it gives me an iterator over references; it does not consume the original iterator. Therefore, I can't pass back the zipped iterators anymore (without also passing back the `grouped_sum` and `grouped_count` data-arrays).

Reply:
Ah ok, no worries then